File: backend/app/models/company.py
from typing import Optional
from datetime import datetime
from sqlmodel import Field, SQLModel


class CompanyProfile(SQLModel, table=True):
    __tablename__ = "company_profiles"
    
    id: Optional[int] = Field(default=None, primary_key=True)
    name: str = Field(default="默认公司")
    target_domains: Optional[str] = None
    budget_range_min: Optional[float] = Field(default=50.0)
    budget_range_max: Optional[float] = Field(default=1000.0)
    qualifications: Optional[str] = None
    service_regions: Optional[str] = None
    bid_history: Optional[str] = None
    is_active: bool = Field(default=True)
    
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)


class CompanyAsset(SQLModel, table=True):
    """Structured company evidence imported from the qualification workbook."""

    __tablename__ = "company_assets"

    id: Optional[int] = Field(default=None, primary_key=True)
    company_name: str = Field(default="")
    asset_type: str = Field(index=True)
    source_sheet: str = Field(index=True)
    name: str = Field(default="", index=True)
    category: Optional[str] = None
    certificate_no: Optional[str] = None
    issuer: Optional[str] = None
    issue_date: Optional[datetime] = None
    expiry_date: Optional[datetime] = None
    status: Optional[str] = None
    amount_wanyuan: Optional[float] = None
    keywords: Optional[str] = None
    data_json: Optional[str] = None
    import_batch_id: Optional[str] = Field(default=None, index=True)
    source_type: str = Field(default="excel_import", index=True)
    is_deleted: bool = Field(default=False, index=True)
    deleted_at: Optional[datetime] = None
    deleted_reason: Optional[str] = None

    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)

File: frontend/src/types/company.ts
export interface CompanyProfile {
  name: string
  target_domains: string[]
  budget_range: [number, number]
  qualifications: string[]
  service_regions: string[]
  bid_history?: BidRecord[]
  asset_summary?: CompanyAssetSummary
  assets?: CompanyAsset[]
}

export interface BidRecord {
  tender_id: string
  tender_title: string
  bid_date: string
  result: 'won' | 'lost' | 'pending'
  feedback?: string
}

export interface CompanyProfileInput {
  name?: string
  target_domains?: string[]
  budget_range?: [number, number]
  qualifications?: string[]
  service_regions?: string[]
}

export interface CompanyAssetSummary {
  total_assets: number
  by_type: Record<string, number>
  by_sheet: Record<string, number>
  by_status: Record<string, number>
  valid_qualification_count: number
  expired_count: number
  expiring_soon_count: number
  top_qualifications: string[]
}

export interface CompanyAsset {
  id: number
  company_name: string
  asset_type: string
  source_sheet: string
  name: string
  category?: string
  certificate_no?: string
  issuer?: string
  issue_date?: string
  expiry_date?: string
  status?: string
  amount_wanyuan?: number
  keywords: string[]
  data: Record<string, unknown>
  import_batch_id?: string
  source_type?: 'excel_import' | 'manual' | 'manual_edit' | string
  is_deleted?: boolean
  deleted_at?: string | null
  deleted_reason?: string | null
}

export interface CompanyImportResult {
  batch_id: string
  company_name: string
  summary: CompanyAssetSummary
  warnings: string[]
}

export interface CompanyImportPreview extends CompanyImportResult {
  preview_id: string
  filename: string
  assets_sample: CompanyAsset[]
}

export interface CompanyAssetsResponse {
  total: number
  skip: number
  limit: number
  items: CompanyAsset[]
}

export interface CompanyAssetQuery {
  asset_type?: string
  status?: string
  source_sheet?: string
  keyword?: string
  include_deleted?: boolean
  skip?: number
  limit?: number
}

export interface CompanyAssetInput {
  company_name?: string
  asset_type: string
  source_sheet?: string
  name: string
  category?: string
  certificate_no?: string
  issuer?: string
  issue_date?: string
  expiry_date?: string
  status?: string
  amount_wanyuan?: number | null
  keywords?: string[]
  data?: Record<string, unknown>
  source_type?: string
}
